java/statestore/src/test/java/sleeper/statestore/dynamodb/DynamoDBFileReferenceStoreIT.java

```diff
@@ -366,28 +375,46 @@ void shouldFailToSplitFileWhichDoesNotExist() throws StateStoreException {
             assertThatThrownBy(() ->
                     store.splitFileReferences(List.of(
                             splitFileToChildPartitions(file, "L", "R"))))
-                    .isInstanceOf(StateStoreException.class);
+                    .isInstanceOf(SplitRequestsFailedException.class)
+                    .cause().isInstanceOf(FileReferenceNotFoundException.class)
+                    .cause().isInstanceOf(TransactionCanceledException.class);
             assertThat(store.getFileReferences()).isEmpty();
             assertThat(store.getAllFileReferencesWithMaxUnreferenced(100))
                     .isEqualTo(noFilesReport());
         }
 
         @Test
-        void shouldFailToSplitFileWhenTheSameFileWasAddedBackToParentPartition() throws StateStoreException {
+        void shouldFailToSplitFileWhenReferenceDoesNotExistInPartition() throws StateStoreException {
+            // Given
+            splitPartition("root", "L", "R", 5);
+            FileReference file = factory.rootFile("file", 100L);
+            FileReference existingReference = splitFile(file, "L");
+            store.addFile(existingReference);
+
+            // When / Then
+            assertThatThrownBy(() ->
+                    store.splitFileReferences(List.of(
+                            splitFileToChildPartitions(file, "L", "R"))))
+                    .isInstanceOf(SplitRequestsFailedException.class)
+                    .cause().isInstanceOf(FileReferenceNotFoundException.class)
+                    .cause().isInstanceOf(TransactionCanceledException.class);
+            assertThat(store.getFileReferences()).containsExactly(existingReference);
+            assertThat(store.getAllFileReferencesWithMaxUnreferenced(100))
+                    .isEqualTo(activeFilesReport(DEFAULT_UPDATE_TIME, existingReference));
+        }
+
+        @Test
+        void shouldFailToAddFileAfterItHasBeenSplit() throws StateStoreException {
```

Comment:
Can this behaviour be applied to the S3 and in-memory state stores as well please?